pkg/acsengine/template_generator.go

```diff
@@ -919,5 +919,8 @@ func (t *TemplateGenerator) getTemplateFuncMap(cs *api.ContainerService) templat
 		},
 		"quote":      strconv.Quote,
 		"shellQuote": helpers.ShellQuote,
+		"GetServicePrincipalClientSecret": func() string {
```

Comment:
This func doesn't appear to handle the `if properties.ServicePrincipalProfile.KeyvaultSecretRef != nil {` case in `pkg/acsengine/params_k8s.go`.

Reply:
I checked the code and the KeyVaultSecretRef part of SPs seems to be used only in the e2e and they don't rely on these template helper methods. So any workflow that involves KeyVault will be unaffected by this change from what I've seen.